src/SearchField/index.jsx

```diff
@@ -0,0 +1,278 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import FontAwesomeStyles from 'font-awesome/css/font-awesome.min.css';
+
+import newId from '../utils/newId';
+import Icon from '../Icon';
+import InputText from '../InputText';
+import Button from '../Button';
+
+import styles from './Search.scss';
+
+const defaultProps = {
+  placeholder: '',
+  value: '',
+  onFocus: () => {},
+  onBlur: () => {},
+  onChange: () => {},
+  hasClearButton: true,
+  hasRoundedCorners: false,
+  searchButtonLabel: 'Search',
+  inputLabel: '',
+  hideInputLabel: false,
+};
+
+const propTypes = {
+  screenReaderText: PropTypes.string.isRequired,
+  placeholder: PropTypes.string,
+  value: PropTypes.string,
+  hideInputLabel: PropTypes.bool,
+  inputLabel: PropTypes.string,
+  searchButtonLabel: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  hasClearButton: PropTypes.bool,
+  hasRoundedCorners: PropTypes.bool,
+  onChange: PropTypes.func,
+  onFocus: PropTypes.func,
+  onBlur: PropTypes.func,
+  onSubmit: PropTypes.func.isRequired,
```

Comment:
Makes a ton of sense to group things alphabetically 👏 
But how do you feel about grouping the required props first (and alphabetically) and the optional props second (and also alphabetically)?

Reply:
👍